homeassistant/components/broadlink/switch.py

```diff
@@ -398,3 +416,135 @@ def _auth(self, retry):
         if not auth and retry > 0:
             return self._auth(retry - 1)
         return auth
+
+
+class BroadlinkBG1Slot(BroadlinkRMSwitch):
+    """Representation of a slot of Broadlink switch."""
+
+    def __init__(self, friendly_name, device, slot, parent_device, retry_times):
+        """Initialize the slot of switch."""
+        super().__init__(friendly_name, friendly_name, device, None, None, retry_times)
+        self._command_on = 1
+        self._command_off = 0
+        self._slot = slot
+        self._parent_device = parent_device
+
+    @property
+    def assumed_state(self):
+        """Return true if unable to access real state of entity."""
+        return False
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        self._turn_on_off(True)
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        self._turn_on_off(False)
+
+    def _turn_on_off(self, state):
+        if self._slot == 1:
+            res = self._device.set_state(pwr1=int(state))
+        else:
+            res = self._device.set_state(pwr2=int(state))
+        if res:
+            _LOGGER.debug("Setting device state: %s", res)
+            self._state = int(state)
+            self._parent_device.set_outlet_status(self._slot, int(state))
+            self.schedule_update_ha_state()
+        else:
+            _LOGGER.warning("No response from switch")
+
+    def _sendpacket(self, packet, retry):
+        """Send packet to device."""
+        try:
+            self._device.set_power(self._slot, packet)
+        except (socket.timeout, ValueError) as error:
```

Comment:
This was duplicated straight from the existing Switch and Slot implementations, so if this is an issue then those need fixing as well? In fact, looking (briefly) again I don't think this method is even used in this class :| 

However, I am missing exception handling and retry in the `_turn_on_off()` method which does basically the same duty :facepalm: - I'll fix that. 

Re auth, the nearest thing I can see from the current implementation is that the `auth` method just catches the generic `OSError`. It would be easy enough to change `socket.timeout` to catch `OSError` instead, which should cover everything, and the polled `update` should handle re-authenticating. Would that work?

Reply:
Yes, this problem is real and it is everywhere. I am working on a solution that uses helper functions to handle exceptions. In the end it will be easier for you. I'll let you know when I finish.